angular-with-directus/how to integrate directus with Angular.md

```diff
@@ -0,0 +1,396 @@
+
+## Serving Dynamic Pages
+In your Directus project, create a new collection named `job` with the fields `id`, `title` , and `description`.
+
+However, the collection should not be a singleton. In addition, grant public access and add some job items to the collection. 
+
+Angular uses path variables and query parameters to serve dynamic data. As a result, you will leverage the `id` as the path variable to display different type of job items.
+
+### Create a Component for Dynamic Pages
+On an existing terminal window of your project, use the following command to generate the the job detail component.
+
+```bash
+ng g c component/job-detail
+```
+Replace the code in the `src/app/component/job-detail/job-detail.component.ts` file with the following code.
+
+```ts
+import {Component, OnInit} from '@angular/core';
+import {directus, Job} from "../../../../directus";
+import {ActivatedRoute} from "@angular/router";
+import {CommonModule, NgIf} from "@angular/common";
+import {readItem} from "@directus/sdk";
+
+@Component({
+  selector: 'app-job-detail',
+  standalone: true,
+  imports: [
+    CommonModule
+  ],
+  templateUrl: './job-detail.component.html',
+  styleUrl: './job-detail.component.css'
+})
+export class JobDetailComponent implements OnInit{
+  job: Job;
+  constructor(private route: ActivatedRoute) {
+  }
+  ngOnInit(): void {
+    this.getJobById(+this
+      .route
+      .snapshot
+      .paramMap.get('id'))
+  }
+  async getJobById(id: number){
+    //@ts-ignore
+    this.job = await directus
+      .request<Job>(readItem("job", id))
+  }
+}
+```
+When the component is initialized, the `id` path variable is retrieved using `ActivatedRoute` and passed to the `readItem()` method to get a job with that id.
+
+The retrieved job item is stored in the object named `job`. To display the contents of the job item, replace the code in the `src/app/component/job-detail/job-detail.component.html` file with the following code.
+
+```html
+<div *ngIf="job">
+  <h2>{{job.title}}</h2>
+  <p>{{job.description}}</p>
+</div>
+```
+### Add Routing for Dynamic Pages
+Go to `src/app/app.routes.ts` file and add the following route in the `Routes` array.
+
+```ts
+  {path: 'job/:id', component: JobDetailComponent},
+```
+when the application reloads, go to `http://localhost:4200/job/id` to view a job item based on its id. Replace the `id` path variable with an appropriate primary key. In addition, replace the id with other keys to view different job items.
+
+## List All the Blog Posts
+In your Directus project, create a collection named `gallery` with the fields `id`, `image`, `title`, and `description`. 
+
+Next, add some items to the collection and grant public access. 
+
+You also need to grant public access to the files to access images. To achieve this, go to **Settings > Access Control > Public > System Collections** and allow read permission for `directus_files`.
```

Comment:
Refer to this for how to explain data modeling and providing sample data: https://docs.directus.io/blog/getting-started-directus-and-eleventy-11ty-3.html#creating-blog-posts-with-directus

Reply:
Okay.